Source/cmNinjaTargetGenerator.cxx

```diff
@@ -333,6 +333,49 @@ std::string cmNinjaTargetGenerator::ComputeIncludes(
   return includesString;
 }
 
+//result - vector of all libraries
+std::vector<std::string> cmNinjaTargetGenerator::ComputeLinkDepsForJSON(
+  const std::string& linkLanguage, const std::string& config,
+  bool ignoreType) const
+{
+  std::vector<std::string> result{};
+  // Static libraries never depend on other targets for linking.
+  if (!ignoreType &&
+      (this->GeneratorTarget->GetType() == cmStateEnums::STATIC_LIBRARY ||
+       this->GeneratorTarget->GetType() == cmStateEnums::OBJECT_LIBRARY)) {
+    return result;
+  }
+
+  cmComputeLinkInformation* cli =
+    this->GeneratorTarget->GetLinkInformation(config);
+  if (!cli) {
+    return result;
+  }
+
+  result = cli->GetDepends();
+
+  // Add a dependency on the link definitions file, if any.
+  if (cmGeneratorTarget::ModuleDefinitionInfo const* mdi =
+    this->GeneratorTarget->GetModuleDefinitionInfo(config)) {
+    for (cmSourceFile const* src : mdi->Sources) {
+      result.push_back(src->GetFullPath());
+    }
+  }
+
+  // Add a dependency on user-specified manifest files, if any.
+  std::vector<cmSourceFile const*> manifest_srcs;
```

Comment:
How does manifest file look like? What is the extension of this file? Do they really need to be added to files field?